Modules/Forum/classes/class.ilForumPostsDeleted.php

```diff
@@ -89,7 +89,16 @@ public function __construct(ilObjForumNotificationDataProvider $provider = NULL)
 		
 		if(is_object($provider))
 		{
-			$this->setDeletedBy($this->user->getLogin());
+			if($provider->objPost->getUserAlias() && $provider->objPost->getDisplayUserId() == 0
```

Comment:
Can you please extract this condition to a method with a meaningful name?

Reply:
done